internal/command/init.go

```diff
@@ -715,8 +715,8 @@ func (c *InitCommand) getProviders(ctx context.Context, config *configs.Config,
 			default:
 				diags = diags.Append(tfdiags.Sourceless(
 					tfdiags.Error,
-					"Failed to query available provider packages",
-					fmt.Sprintf("Could not retrieve the list of available versions for provider %s: %s",
+					"Failed to resolve provider packages",
```

Comment:
Question: do we want to add an explanation on how people can fix this?

Reply:
There are a variety of different scenarios that can hit this case.  I think this whole error handler should be investigated, improved, and properly tested in a future PR.